packages/core/src/components/fields/SchemaField.js

```diff
@@ -207,6 +209,8 @@ function WrapIfAdditional(props) {
             <LabelInput
```

Comment:
Remove LabelInput, and replace with

```jsx
<SchemaField
      schema={propertyNamesSchema}
      idSchema={{ $id: "${id}-key" }}
      defaultValue={label}
      registry={props.registry}
      onBlur={(event, value) => onKeyChange(value)}
/>
```

Add a comment saying something like "this lets us render an arbitrary field based on property names (such as dropdown for `enum`)"